src/main/java/ch/njol/skript/effects/EffReplace.java

```diff
@@ -37,57 +37,69 @@
 import org.bukkit.inventory.ItemStack;
 import org.eclipse.jdt.annotation.Nullable;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
 import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 @Name("Replace")
-@Description("Replaces all occurrences of a given text with another text. Please note that you can only change variables and a few expressions, e.g. a <a href='../expressions.html#ExprMessage'>message</a> or a line of a sign.")
-@Examples({"replace \"<item>\" in {textvar} with \"%item%\"",
-		"replace every \"&\" with \"§\" in line 1",
-		"# The following acts as a simple chat censor, but it will e.g. censor mass, hassle, assassin, etc. as well:",
-		"on chat:",
-		"	replace all \"kys\", \"idiot\" and \"noob\" with \"****\" in the message",
-		" ",
-		"replace all stone and dirt in player's inventory and player's top inventory with diamond"})
-@Since("2.0, 2.2-dev24 (replace in multiple strings and replace items in inventory), 2.5 (replace first, case sensitivity)")
+@Description("Replaces all occurrences of a given text/regex with another text. Please note that you can only change " +
+	"variables and a few expressions, e.g. a <a href='/expressions.html#ExprMessage'>message</a> or a line of a sign.")
+@Examples({
+	"replace \"<item>\" in {_msg} with \"[%name of player's tool%]\"",
+	"replace every \"&\" with \"§\" in line 1 of targeted block",
+	"",
+	"# Very simple chat censor",
+	"on chat:",
+	"\treplace all \"kys\", \"idiot\" and \"noob\" with \"****\" in the message",
+	"\treplace using regex \"\\b(kys|idiot|noob)\\b\" with \"****\" in the message # Regex version for better results",
+	"",
+	"replace all stone and dirt in player's inventory and player's top inventory with diamond"
+})
+@Since("2.0, 2.2-dev24 (multiple strings, items in inventory), 2.5 (replace first, case sensitivity), INSERT VERSION (regex)")
 public class EffReplace extends Effect {
 
 	static {
 		Skript.registerEffect(EffReplace.class,
-				"replace (all|every|) %strings% in %strings% with %string% [(1¦with case sensitivity)]",
-				"replace (all|every|) %strings% with %string% in %strings% [(1¦with case sensitivity)]",
-				"replace first %strings% in %strings% with %string% [(1¦with case sensitivity)]",
-				"replace first %strings% with %string% in %string% [(1¦with case sensitivity)]",
+				"replace (all:(all|every)|first:[the] first|) %strings% in %strings% with %string% [(case:with case sensitivity)]",
+				"replace (all:(all|every)|first:[the] first|) %strings% with %string% in %strings% [(case:with case sensitivity)]",
+				"regex:(replace [using] regex|regex replace) %strings% in %strings% with %string%",
+				"regex:(replace [using] regex|regex replace) %strings% with %string% in %strings%",
 				"replace (all|every|) %itemtypes% in %inventories% with %itemtype%",
 				"replace (all|every|) %itemtypes% with %itemtype% in %inventories%");
```

Comment:
While we're here, we can do some minor pattern cleanup
```suggestion
				"replace [all:(all|every)|first:[the] first] %strings% in %strings% with %string% [case:with case sensitivity]",
				"replace [all:(all|every)|first:[the] first] %strings% with %string% in %strings% [case:with case sensitivity]",
				"regex:(replace [using] regex|regex replace) %strings% in %strings% with %string%",
				"regex:(replace [using] regex|regex replace) %strings% with %string% in %strings%",
				"replace [all|every] %itemtypes% in %inventories% with %itemtype%",
				"replace [all|every] %itemtypes% with %itemtype% in %inventories%");
```